File: leetcode/two-pointers/11.cpp
#include<bits/stdc++.h>
using namespace std;
int maxArea(vector<int>& height) {
    
    if(height.size() == 1) return height[0];
    int l = 0, r = height.size()-1, res, max = -1;
    while(l < r)
    {
        if(height[l] < height[r])
            l++;
        else
            r--;
        res = min(height[l], height[r]) * abs(r - l);
        if(res > max)
        {
            max = res;
        }           
    }
    
        
    return max;
}
int main()
{
    vector<int> height;
    //[1,3,2,25,24,5]
    height.push_back(1);
    height.push_back(8);
    height.push_back(6);
    height.push_back(2);
    height.push_back(5);
    height.push_back(4);
    height.push_back(8);
    

    cout << maxArea(height);
}

File: leetcode/Go/two-pointers/11/11.go
package main

func maxArea(height []int) int {
	p1, p2 := 0, len(height)-1
	max := 0
	for p1 < p2 {
		width := p2 - p1
		h := min(height[p1], height[p2])
		area := width * h
		if area > max {
			max = area
		}
		if height[p1] < height[p2] {
			p1++
		} else {
			p2--
		}
	}
	return max
}
